lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -496,13 +508,16 @@ def archive_export(self, trans, id, **kwds):
             # written.
             job_id = trans.security.encode_id(job.id)
             return dict(job_id=job_id)
+
         if up_to_date and jeha.ready:
-            jeha_id = trans.security.encode_id(jeha.id)
-            return dict(download_url=url_for("history_archive_download", id=id, jeha_id=jeha_id))
+            return self.history_export_view.serialize(trans, id, jeha)
         else:
             # Valid request, just resource is not ready yet.
             trans.response.status = "202 Accepted"
-            return ''
+            if jeha:
+                return self.history_export_view.serialize(trans, id, jeha)
+            else:
+                return ''
```

Comment:
That's what's being returned on the first click when exporting a history to link, while the UI is waiting for a job id (`this.waitOnExportJob(response.data.job_id)`). That fails, so then there is just a console error and the view never updates.

I think we can return the encoded job id as on line 510, that shouldn't break older code that polls for a download link

Reply:
This was a last minute change to get the tests working again, I didn't revisit the UI - sorry about that. I'll try to resolve this.

Reply:
I've made this change!